File: app/src/main/java/com/musicplayer/SocyMusic/ui/playlist/PlaylistSongAdapter.java
package com.musicplayer.SocyMusic.ui.playlist;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.musicplayer.SocyMusic.data.Song;
import com.musicplayer.musicplayer.R;

import java.util.List;

public class PlaylistSongAdapter extends RecyclerView.Adapter<PlaylistSongHolder> {
    private Context context;

    private List<Song> songs;
    private ItemClickListener clickListener;

    public PlaylistSongAdapter(Context context, List<Song> songs) {
        this.context = context;
        this.songs = songs;
    }

    @NonNull
    @Override
    public PlaylistSongHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        View itemView = inflater.inflate(R.layout.list_item_playlist_song, parent, false);
        return new PlaylistSongHolder(itemView, clickListener);
    }

    @Override
    public void onBindViewHolder(@NonNull PlaylistSongHolder holder, int position) {
        holder.bind(songs.get(position));
    }

    @Override
    public int getItemCount() {
        return songs.size();
    }

    public void setSongs(List<Song> songs) {
        this.songs = songs;
    }

    public void setOnItemClickListener(ItemClickListener clickListener) {
        this.clickListener = clickListener;
    }


    public interface ItemClickListener {
        void onItemClick(int position, View view);

        boolean onItemLongClick(int position, View view);
    }

}

File: app/src/main/kotlin/com/musicplayer/openmusic/ui/playlist/PlaylistSongAdapter.kt
package com.musicplayer.openmusic.ui.playlist

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.musicplayer.openmusic.data.Song
import com.musicplayer.musicplayer.R

class PlaylistSongAdapter(private val context: Context, private var songs: List<Song>) :
    RecyclerView.Adapter<PlaylistSongHolder>() {
    private var clickListener: ItemClickListener? = null
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): PlaylistSongHolder {
        val inflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
        val itemView = inflater.inflate(R.layout.list_item_playlist_song, parent, false)
        return PlaylistSongHolder(itemView, clickListener!!)
    }

    override fun onBindViewHolder(holder: PlaylistSongHolder, position: Int) {
        holder.bind(songs[position])
    }

    override fun getItemCount(): Int {
        return songs.size
    }

    fun setSongs(songs: List<Song>) {
        this.songs = songs
    }

    fun setOnItemClickListener(clickListener: ItemClickListener) {
        this.clickListener = clickListener
    }

    interface ItemClickListener {
        fun onItemClick(position: Int, view: View)
        fun onItemLongClick(position: Int, view: View): Boolean
    }
}